cosmic_ray/operators/operator.py

```diff
@@ -53,3 +54,23 @@ def __repr__(self):
         return '{}({})'.format(
             self.__class__.__name__,
             ', '.join(args))
+
+
+class ReplacementOperator(Operator, metaclass=ABCMeta):
+    """Base class for operators that have a "from-operator" and a "to-operator".
+
+    This is primarily a mechanism to tell linters that these properties exist.
+    Several of our operators build subclasses dynamically, and they primarily
+    just configure different to/from-ops.
+    """
+    @property
+    @abstractmethod
+    def from_op(self):
+        """The operator to mutate from."""
+        pass
```

Comment:
Consider `raise NotImplementedError` unless you intend to return `None` in the case this implementation is called.?